src/nvim/lua/vim.lua

```diff
@@ -386,4 +386,64 @@ do
   vim.wo = new_win_opt_accessor(nil)
 end
 
+--- Get table of lines with start, end columns for given marks
+---
+-- @param mark1 mark of beginning of range
+-- @param mark2 mark of end of range
+-- @param regtype type of selection that is yanked (:help setreg)
+-- @param boolean indicating whether the selection is end-inclusive
+function vim.marks_to_region(mark1, mark2, regtype, inclusive)
```

Comment:
can't think of a reason not to name it simply `vim.region()`. Most of our Lua
functions have terse names (except for stuff in LSP, but hopefully we can
revisit that before 0.5 is released...).

And I do not see why this should be restricted to marks. That's why I suggested
`pos1`, `pos2`. Is there a reason not to accept (row,col) pairs for `pos{1,2}` ?

Reply:
Nit: all code in the project uses 2-space indent

Reply:
Because the logic is very specific to extracting the necessary information from the `getpos` output?

Reply:
Everything after 399 is generic to any pair of `(row,col)` coordinates + region
`type` + `inclusive` flag.

Reply:
That's what I meant below.

Reply:
(`type` is a reserved keyword in lua.)